devices/uartnmeagps.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+
+import calendar
+import datetime
+import dotenv
+import io
+import os
+import re
+import time
+import threading
+
+from absl import logging
+from serial import Serial
+from pynmeagps import NMEAReader 
+from . import Sensor
+
+
+class UartNmeaGps(Sensor):
+  def __init__(self, remotestorage, localstorage, timesource, interval=None, send_last_known_gps=False, env_file=None, **kwargs):
+    super().__init__(remotestorage, localstorage, timesource)
+    # Stream represents our connection to the UART.
+    self.stream = None
+    self.last_error = ''
+
+    # If available, we will save last known GPS coordinates to environment variables.
```

Comment:
Does this save to disk every time?

Reply:
Yep.  Whenever possible, we save the last-known GPS coordinates.